src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs

```diff
@@ -113,6 +113,42 @@ public void WriteRawValue(ReadOnlySpan<byte> utf8Json, bool skipInputValidation
             WriteRawValueCore(utf8Json, skipInputValidation);
         }
 
+        /// <summary>
+        /// Writes the input as JSON content. It is expected that the input content is a single complete JSON value.
+        /// </summary>
+        /// <param name="utf8Json">The raw JSON content to write.</param>
+        /// <param name="skipInputValidation">Whether to validate if the input is an RFC 8259-compliant JSON payload.</param>
+        /// <exception cref="ArgumentException">Thrown if the length of the input is zero or equal to <see cref="int.MaxValue"/>.</exception>
+        /// <exception cref="JsonException">
+        /// Thrown if <paramref name="skipInputValidation"/> is <see langword="false"/>, and the input
+        /// is not a valid, complete, single JSON value according to the JSON RFC (https://tools.ietf.org/html/rfc8259)
+        /// or the input JSON exceeds a recursive depth of 64.
+        /// </exception>
+        /// <remarks>
+        /// When writing untrused JSON values, do not set <paramref name="skipInputValidation"/> to <see langword="true"/> as this can result in invalid JSON
+        /// being written, and/or the overall payload being written to the writer instance being invalid.
+        ///
+        /// When using this method, the input content will be written to the writer destination as-is, unless validation fails (when it is enabled).
+        ///
+        /// The <see cref="JsonWriterOptions.SkipValidation"/> value for the writer instance is honored when using this method.
+        ///
+        /// The <see cref="JsonWriterOptions.Indented"/> and <see cref="JsonWriterOptions.Encoder"/> values for the writer instance are not applied when using this method.
+        /// </remarks>
+        public void WriteRawValue(ReadOnlySequence<byte> utf8Json, bool skipInputValidation = false)
+        {
+            if (!_options.SkipValidation)
+            {
+                ValidateWritingValue();
+            }
+
+            if (utf8Json.Length == int.MaxValue)
```

Comment:
I know this check has been taken from the `ReadOnlySpan` overload, but I struggle to understand what it means. Why only check for `int.MaxValue` lengths specifically?

For the case of `ReadOnlySpan`, it will almost always be false (as managed buffers are bounded by `Array.MaxLength` which in turn is less than `int.MaxValue`). For the case of `ReadOnlySequence` the `Length` property is `long` so any value `> int.MaxValue` will happily pass this check.

Reply:
cc @layomia

Reply:
Fixed.
For `ReadOnlySequence `overload side, it is my oversight when I thought they are similar codes at first sight, sorry about that ...
For `ReadOnlySpan `overload side, I guess this int length cannot be int.MaxValue because [subsequent code](https://github.com/dotnet/runtime/blob/1281a4a3393e9293aa4b379d619a692d6dcb62a4/src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs#L182) may add `','`.

Reply:
Ok, just confirm, we will see @layomia 's reply about that check.

Reply:
This is the only remaining question on this PR--whether the check for `len == int.MaxValue` is a valuable defense against an overflow exception.

@lateapexearlyspeed To determine if the guard is useful or not, could you try adding a unit test that would try to hit that boundary case? Let's see what happens with and without the guard. A similar test could be added for the `ReadOnlySpan` overload too.

Reply:
@lateapexearlyspeed I would make the following recommendations:

1. Completely remove the `len == int.MaxValue` check (and update API docs above) from the existing span overload.
2. Update the check in the new ReadOnlySequence overload to say `len <= int.MaxValue` (so that the length can be safely cast to an `int32`).
3. Update the assertion in [the core methods](https://github.com/dotnet/runtime/blob/a715f30d106f69e6243c46f82a3cacd12568936d/src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs#L237-L241) to say `len <= int.MaxValue`.
4. Add unit tests covering these boundary conditions.

Thanks!

Reply:
> Add unit tests covering these boundary conditions

I believe these are covered here but it's still worth some thought https://github.com/dotnet/runtime/blob/af8694704e807d0147a3033c3e250be2966951fe/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.WriteRaw.cs#L478

Reply:
For new ROSeq, there was also [test case](https://github.com/dotnet/runtime/blob/a715f30d106f69e6243c46f82a3cacd12568936d/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.WriteRaw.cs#L532) added to cover boundary conditions.

Reply:
> It wasn't.
> I recommend we leave as-is. We require the the input length is than `int.MaxValue` so we can add a comma if required. The comment talks about validation in the Utf16 based entry points (the info about integer division) but should also state that the ROSpan/Sequence entry points need to adhere to this to.

I'm not sure I follow. Even you are able to add one comma at the end, that isn't necessary to produce a complete JSON document. 

My concern is that in 99.9% of use cases, this check isn't particularly meaningful since both the source and `Utf8JsonWriter` destination buffers are bounded by `Array.MaxLength` which is substantially less than `int.MaxValue`. You could, in theory, construct an unmanaged span of length `int.MaxValue`, but if you do that then your code probably has bigger problems than not being able to append one character at the end.

In any case, I probably agree we should not try to change this in the context of this PR.

Reply:
> Utf8JsonWriter destination buffers are bounded by Array.MaxLength which is substantially less than int.MaxValue.

Yeah that's right - context I didn't keep in mind when thinking about this.

> we should not try to change this in the context of this PR.

👍 yeah fine to address in a follow up.